### PR TITLE
Make blur input scaling adaptive by default

### LINKED ISSUE #1083: Explore adaptive input scaling for blur performance
## Motivation

Android macrobenchmark and Perfetto analysis show that blur cost is dominated by RenderThread/Vulkan work over the retained layer. Reducing the blur input resolution is the lowest-risk lever found so far for improving the main benchmark.

## Evidence

- `HazeInputScale.Auto` improved macrobenchmark CPU by approximately 0.4–0.5 ms.
- A representative Perfetto trace improved actual frame P90 by approximately 1.27 ms.
- The scaled path reduced render operations, but retained the extra Haze render submission, so this is an incremental optimization rather than a complete solution.

## Explore

- Evaluate whether adaptive input scaling should become the default for sufficiently large blur surfaces or radii.
- Compare `Auto` with a less aggressive fixed scale such as `0.5f`.
- Consider thresholds based on blur radius, layer dimensions, device density, or pixel count.
- Preserve explicit caller-provided input scaling.
- Check whether the policy should be Android-specific or consistent across platforms.

## Success criteria

- Paired macrobenchmark runs show a repeatable improvement in the main scaffold benchmark.
- Screenshot and perceptual comparisons show acceptable blur quality.
- No regression in progressive blur geometry or cross-platform rendering invariants.
- Document the selected policy and its visual/performance tradeoff.

### PR DESCRIPTION
**Pull request by chrisbanes**
## Summary

- add a distinct effect-default input-scale state so Blur can adapt automatically while Glass remains full resolution by default
- select Blur input scale from physical blur radius and expanded layer area using 1.0, 0.8, and 0.5 tiers with hysteresis and a progressive-blur cap
- reset adaptive hysteresis on effect detach and delegate changes so state cannot leak across attachment owners or scrim intervals
- preserve explicit None, Auto, and Fixed caller control; add common policy tests, lifecycle coverage, Android and Desktop screenshots, benchmark routes, API docs, migration guidance, changelog, and an ADR
- remove the former automatic 0.3334 tier

## Compatibility

Adding `HazeInputScale.EffectDefault` extends a public sealed hierarchy. This is a deliberate source and runtime breaking change and requires a breaking release. The migration guide and changelog call this out explicitly.

## Performance evidence

Pixel 6, 90 Hz, 16 iterations per scenario, three interleaved pairs:

| Comparison | Median CPU P90 | Median actual-frame P90 |
| --- | ---: | ---: |
| Unscaled vs adaptive | about -0.73% | -4.21% |
| Fixed 0.8 vs adaptive | +0.41% | -2.51% |

Adaptive improved actual-frame P90 in every pair. The fixed-0.8 comparison shows the extra 0.5 tier primarily reduces RenderThread and Vulkan work; CPU P90 was mixed. Progressive and masked scenarios improved actual-frame P90 by 5.76% and 6.53% respectively. The ADR records all pair-level results and methodology.

## Validation

- `./gradlew spotlessCheck`
- `./gradlew :haze:check`
- `./gradlew :haze-blur:check :haze-glass:check`
- `./gradlew :sample:android:assembleDebug :sample:desktop:compileKotlinJvm`
- `./gradlew :internal:benchmark:assembleBenchmark`
- Android and Desktop Roborazzi verification
- Metalava compatibility and Dokka generation
- final reuse, quality, efficiency, clarity, ponytail, Standards, and Spec reviews clean on `8fd484ba`

Fixes #1083

### CODE REVIEW

File: haze-blur/src/commonMain/kotlin/dev/chrisbanes/haze/blur/BlurVisualEffect.kt [hunk 1]
```diff
@@ -69,6 +68,7 @@ public class BlurVisualEffect() : VisualEffect, RetainedOutputVisualEffect {
   private var isAttached: Boolean = false
 
   private var needsDelegateSelection: Boolean = true
+  private val inputScalePolicy = BlurInputScalePolicy()
```
[the remark this answers is not in the document]
Comment (chrisbanes) [reply haze-blur/src/commonMain/kotlin/dev/chrisbanes/haze/blur/BlurVisualEffect.kt:71]: Fixed in 870842f3. `BlurVisualEffect.detach()` now resets the adaptive policy after delegate teardown, so hysteresis cannot cross attachment owners. Added `reset_clearsAutomaticHysteresisHistory` as a focused regression test; `spotlessCheck`, `:haze-blur:check`, and `:haze:check` pass.

[the remark this answers is not in the document]
Comment (chrisbanes) [reply haze-blur/src/commonMain/kotlin/dev/chrisbanes/haze/blur/BlurVisualEffect.kt:71]: Fixed in 870842f3. `BlurVisualEffect.detach()` now resets the adaptive policy after delegate teardown, so hysteresis cannot cross attachment owners. Added `reset_clearsAutomaticHysteresisHistory` as a focused regression test; `spotlessCheck`, `:haze-blur:check`, and `:haze:check` pass.

**Comment by chrisbanes** [comment]
@codex review

**Comment by chrisbanes** [comment]
@codex review

**Comment by chrisbanes** [comment]
Addressed the second Greptile finding in 8fd484ba. Adaptive history now resets whenever the blur delegate changes, covering blur-to-scrim-to-blur transitions, with `changingDelegate_resetsAutomaticHysteresisHistory` reproducing the stale 0.5 tier before the fix. Spotless, `:haze-blur:check`, `:haze:check`, and the complete exact-SHA review gate pass.

**Comment by chrisbanes** [comment]
@codex review
